gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -694,42 +712,59 @@ object InMemoryTableScanHelper {
 }
 
 // This rule will try to add RowToColumnarExecBase and ColumnarToRowExec
-// to support vanilla columnar scan.
+// to support vanilla columnar operators.
```

Comment:
I'm afraid the scope is bigger than Gluten: we are trying to be compatible with vanilla columnar operators which is not Spark built-in operators.

Reply:
This change is for supporting "vanilla columnar operators which is not Spark built-in operators.". I think Gluten should support this or declare this as a limitation. Please let me know if any concern for supporting it.
Now gluten logic wrpper all columnar ops (except readers) with `VeloxColumnarToRowExec` and `VeloxRowToColumnarExec` and caused the exceptions. This PR just fix the behavior to make sure, only gluten plan is wrappered with velox version C2R and R2C. other columnar plans are still wrppered with vanilla version C2R and R2C.